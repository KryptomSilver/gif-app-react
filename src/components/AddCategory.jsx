import React, { useState } from "react";
import PropTypes from "prop-types";
const AddCategory = ({ setCategories }) => {
  const [input, setInput] = useState("");
  const handleInputChange = (e) => {
    setInput(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (input.trim().length > 2) {
      setCategories((items) => [input, ...items]);
      setInput("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{input}</p>
      <input type="text" onChange={handleInputChange} value={input} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
export default AddCategory;
