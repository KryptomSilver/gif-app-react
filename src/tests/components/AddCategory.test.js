import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";
import React from "react";

describe("Debe de evaluar el componente <AddCategory/>", () => {
  const setCategories = jest.fn();
  let wraper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wraper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test("Debe de mostrar el componente correctamente ", () => {
    expect(wraper).toMatchSnapshot();
  });
  test("Debe de cambiar la caja de texto", () => {
    const input = wraper.find("input");
    const value = "Hola mundo";
    input.simulate("change", { target: { value } });
    expect(wraper.find("p").text().trim()).toBe(value);
  });
  test("no debe de postear al informaciÃ³n onSubmit", () => {
    wraper.find("form").simulate("submit", {
      preventDefault() {},
    });
    expect(setCategories).not.toHaveBeenCalled();
  });
  test("Debe de llamar el setCategories y limpiar la caja de texto", () => {
    // 1 Simular el imputChange
    const input = wraper.find("input");
    const value = "Cambio";
    input.simulate("change", { target: { value } });
    expect(wraper.find("p").text().trim()).toBe(value);
    // 2 Simular el submit
    wraper.find("form").simulate("submit", {
      preventDefault() {},
    });
    // 3 setCategories se debe de haber llamado
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    // 4 El valor del input debe estar en ""
    expect(wraper.find("input").prop("value")).toBe("");
  });
});
