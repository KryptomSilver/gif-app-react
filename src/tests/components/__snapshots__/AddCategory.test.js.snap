// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de evaluar el componente <AddCategory/> Debe de mostrar el componente correctamente  1`] = `ShallowWrapper {}`;
